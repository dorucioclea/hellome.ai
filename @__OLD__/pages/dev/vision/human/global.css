@font-face {
  font-style: normal;
  font-weight: 100;
  src: local("Lato"), url("/lato-light.woff2");
  font-family: "Lato";
  font-display: swap;
}
html {
  font-variant: small-caps;
  font-size: 16px;
  font-family: "Lato", "Segoe UI";
}
body {
  margin: 0;
  background: black;
  width: 100vw;
  height: 100vh;
  overflow-x: hidden;
  color: white;
}
body::-webkit-scrollbar {
  display: none;
}
a {
  color: inherit;
  text-decoration: none;
}
