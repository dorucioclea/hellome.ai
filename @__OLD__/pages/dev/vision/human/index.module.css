/* .container {
  padding: 0 2rem;
} */
.footer {
  display: flex;
  justify-content: center;
  align-items: center;
  border-top: 1px solid #eaeaea;
  padding: 1rem;
}
.webcam {
  display: none;
}
.output {
  margin: 0 auto;
  width: 100%;
}
.status {
  position: absolute;
  top: 12px;
  right: 20px;
  box-shadow: 2px 2px black;
  background-color: grey;
  padding: 8px;
  font-family: monospace;
}
.log {
  border-top: 1px solid #eaeaea;
  padding: 8px;
}
.performance {
  position: absolute;
  bottom: 0;
  padding: 8px;
  width: 100%;
  font-size: 0.8rem;
  font-family: monospace;
}
