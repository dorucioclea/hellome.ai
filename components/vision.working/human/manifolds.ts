import type { FaceResult, Result } from "@vladmandic/human"

import {
  distanceMetric,
  withinConfidenceBounds,
  withinToleranceBounds,
} from "@/components/vision.working/human/metrics"

export type UsableFaceResult = FaceResult & {
  rotation: NonNullable<FaceResult["rotation"]>
}

export type Point2D<T = number> = {
  x: T // pitch
  y: T // yaw
  z: T // roll
}

export type ManifoldCoordinates = {
  axis: Point2D<number[]>
  tolerance: Point2D
  confidence: {
    min: {
      face: number
    }
  }
}

export type ManifoldVector = {
  label: string
  index: Point2D
  origin: Point2D
  tolerance: ManifoldCoordinates["tolerance"]
  candidate?: {
    result: Result
    face: FaceResult
    score: number
  }
}

export function castManifoldVectors({
  axis,
  tolerance,
}: ManifoldCoordinates): ManifoldVector[] {
  return axis.x.flatMap((x, xIndex) => {
    return axis.y.flatMap((y, yIndex) => {
      return axis.z.flatMap((z, zIndex) => {
        const vector: ManifoldVector = {
          label: `x:${x * 100}° y:${y * 100}° z:${z * 100}°`,
          index: { x: xIndex, y: yIndex, z: zIndex },
          origin: { x, y, z },
          tolerance,
        }
        return [vector]
      })
    })
  })
}

export function findNearestManifoldVector(
  coordinates: ManifoldCoordinates,
  vectors: ManifoldVector[],
  face: UsableFaceResult,
): ManifoldVector | undefined {
  if (vectors.length === 0 || !withinConfidenceBounds(face, coordinates)) {
    return undefined
  }

  const toleratedVectors = vectors.filter((vector) =>
    withinToleranceBounds(face, vector),
  )

  if (toleratedVectors.length === 0) {
    return undefined
  }

  return toleratedVectors.reduce((prev, current) => {
    return distanceMetric(face, prev) < distanceMetric(face, current)
      ? current
      : prev
  })
}
