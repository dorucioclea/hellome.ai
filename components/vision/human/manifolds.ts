import type { FaceResult, Result } from "@vladmandic/human"

import {
  distanceMetric,
  withinConfidenceBounds,
  withinToleranceBounds,
} from "@/components/vision/human/metrics"

export type UsableFaceResult = FaceResult & {
  rotation: NonNullable<FaceResult["rotation"]>
}

export type Point2D<T = number> = {
  x: T // pitch
  y: T // yaw
  z: T // roll
}

export type ManifoldCoordinates = {
  axis: Point2D<number[]>
  tolerance: Point2D
  confidence: {
    min: {
      face: number
    }
  }
}

export type ManifoldVector<T, V> = {
  label: string
  index: Point2D
  origin: Point2D
  tolerance: ManifoldCoordinates["tolerance"]
  candidate?: ManifoldVectorCandidate<T, V>
}

export type ManifoldVectorCandidate<T, V> = {
  imageData?: ImageData | undefined | null
  result: T
  face: V
  score: number
}

export function castManifoldVectors<T, V>({
  axis,
  tolerance,
}: ManifoldCoordinates): ManifoldVector<T, V>[] {
  return axis.x.flatMap((x, xIndex) => {
    return axis.y.flatMap((y, yIndex) => {
      return axis.z.flatMap((z, zIndex) => {
        const vector: ManifoldVector<T, V> = {
          label: `x:${x * 100}° y:${y * 100}° z:${z * 100}°`,
          index: { x: xIndex, y: yIndex, z: zIndex },
          origin: { x, y, z },
          tolerance,
        }
        return [vector]
      })
    })
  })
}

export function findNearestManifoldVector(
  coordinates: ManifoldCoordinates,
  vectors: ManifoldVector<Result, FaceResult>[],
  face: UsableFaceResult,
): ManifoldVector<Result, FaceResult> | undefined {
  if (vectors.length === 0 || !withinConfidenceBounds(face, coordinates)) {
    return undefined
  }

  const toleratedVectors = vectors.filter((vector) =>
    withinToleranceBounds(face, vector),
  )

  if (toleratedVectors.length === 0) {
    return undefined
  }

  return toleratedVectors.reduce((prev, current) => {
    return distanceMetric(face, prev) < distanceMetric(face, current)
      ? current
      : prev
  })
}
