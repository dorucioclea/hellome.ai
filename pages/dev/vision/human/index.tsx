import { useRef } from "react"
import Head from "next/head"

import InitWebcam from "@/components/vision/InitWebcam"
import FacePredictionOverlay from "@/components/vision/human/FacePredictionOverlay/FacePredictionOverlay"

import styles from "@/pages/dev/vision/human/index.module.css"

// import { Layout } from '@/components/common'

export default function Index() {
  const canvasRef = useRef<HTMLCanvasElement>(null)
  const videoRef = useRef<HTMLVideoElement>(null)
  const statusRef = useRef<HTMLDivElement>(null)
  const logRef = useRef<HTMLDivElement>(null)
  const performanceRef = useRef<HTMLDivElement>(null)

  return (
    <div className={styles.container}>
      <Head>
        <title>Human</title>
        <meta
          name="description"
          content="Human: Demo with TypeScript/ReactJS/NextJS"
        />
        <link rel="icon" href="/favicon.ico" />
      </Head>

      {/* placeholder element that will be used by human for output */}
      <canvas ref={canvasRef} id="canvas" className={styles.output} />
      <video
        ref={videoRef}
        id="video"
        className={styles.webcam}
        autoPlay
        muted
      />

      {/* <Webcam id="video" className={styles.webcam} autoPlay muted /> */}

      {/* placeholder element that will be used by webcam */}
      <div ref={statusRef} id="status" className={styles.status} />
      <div ref={logRef} id="log" className={styles.log} />
      <div
        ref={performanceRef}
        id="performance"
        className={styles.performance}
      />
      <InitWebcam elementId="video" />

      {/* initialized webcam using html video element with specified id */}
      <FacePredictionOverlay
        inputId="video"
        outputId="canvas"
        canvasRef={canvasRef}
        videoRef={videoRef}
        statusRef={statusRef}
        logRef={logRef}
        performanceRef={performanceRef}

        // handleResult={handleHumanResult}
      />
    </div>
  )
}

// Index.Layout = Layout
