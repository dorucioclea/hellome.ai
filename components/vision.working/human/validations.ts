import { Result } from "@vladmandic/human"

import { ManifoldCoordinates } from "@/components/vision.working/human/manifolds"

export function validateResult(
  coordinates: ManifoldCoordinates,
  result: Result,
): boolean {
  if (result.face.length === 0) {
    // console.warn(`Found no faces`, result)
    return false
  }
  if (result.face.length > 1) {
    console.warn(`Found multiple faces: ${result.face.length}`, result)
    return false
  }
  return true
}

export function isUsable(
  coordinates: ManifoldCoordinates,
  result: Result,
): boolean {
  if (result.face.length === 0) {
    // console.warn(`Found no faces`, result)
    return false
  }
  if (result.face.length > 1) {
    console.warn(`Found multiple faces: ${result.face.length}`, result)
    return false
  }
  if (!result.face[0].rotation) {
    console.warn(`Found no face.rotation`, result)
    return false
  }
  return true
}
