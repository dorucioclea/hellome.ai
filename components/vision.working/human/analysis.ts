import type { Result } from "@vladmandic/human"
import { indexOf } from "lodash"

import {
  ManifoldCoordinates,
  ManifoldVector,
  UsableFaceResult,
  findNearestManifoldVector,
} from "./manifolds"
import { scoreFaceResult } from "./metrics"
import { isUsable } from "./validations"

export type FaceAnalysis = {
  coordinates: ManifoldCoordinates
  vectors: ManifoldVector[]
}

export type FaceAnalysisResult = {
  disposeResults: Result[]
  analysis: FaceAnalysis
}

export function runFaceAnalysis(
  result: Result,
  prevAnalysis: FaceAnalysis,
): FaceAnalysisResult {
  if (!isUsable(prevAnalysis.coordinates, result)) {
    return {
      disposeResults: [result],
      analysis: prevAnalysis,
    }
  }

  // const {coordinates, vectors}= prevAnalysis

  const face = result.face[0] as UsableFaceResult

  const prevVector = findNearestManifoldVector(
    prevAnalysis.coordinates,
    prevAnalysis.vectors,
    face,
  )
  const index = indexOf(prevAnalysis.vectors, prevVector)

  const nextVectors = prevAnalysis.vectors
  if (prevVector && index) {
    const score = scoreFaceResult(face, prevVector)

    if (score > (prevVector.candidate?.score ?? Number.NEGATIVE_INFINITY)) {
      const nextVector = {
        ...prevVector,
        candidate: {
          result,
          face,
          score,
        },
      }

      nextVectors[index] = nextVector
    }

    return {
      disposeResults: prevVector.candidate?.result
        ? [prevVector.candidate.result]
        : [],
      analysis: {
        ...prevAnalysis,
        vectors: nextVectors,
      },
    }
  }

  return {
    disposeResults: [result],
    analysis: prevAnalysis,
  }
}
