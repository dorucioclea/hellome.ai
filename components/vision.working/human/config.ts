import { type FaceAnalysis } from "@/components/vision.working/human/analysis"
import {
  castManifoldVectors,
  type ManifoldCoordinates,
} from "@/components/vision.working/human/manifolds"

export const DEFAULT_COORDINATES: ManifoldCoordinates = {
  axis: {
    x: [-0.45, -0.3, -0.15, 0, 0.15, 0.3, 0.45],
    y: [-0.3, -0.15, 0, 0.15, 0.3],
    z: [0],
  },
  tolerance: {
    x: 0.1,
    y: 0.1,
    z: 0.5,
  },
  confidence: {
    min: {
      face: 0.7,
    },
  },
}

export const defaultFaceAnalysis: FaceAnalysis = {
  coordinates: DEFAULT_COORDINATES,
  vectors: castManifoldVectors(DEFAULT_COORDINATES),
}
