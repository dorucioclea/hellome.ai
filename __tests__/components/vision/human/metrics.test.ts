import { FaceResult, Result } from "@vladmandic/human"
import { range } from "lodash"

import { castManifoldVectors } from "@/components/vision/human/manifolds"
import {
  confidenceMetric,
  distanceMetric,
  scoreFaceResult,
  withinConfidenceBounds,
  withinToleranceBounds,
} from "@/components/vision/human/metrics"

import { DEFAULT_COORDINATES, DEFAULT_USABLE_FACE_RESULT } from "./fixtures"

describe("DEFAULT_COORDINATES", () => {
  const face = DEFAULT_USABLE_FACE_RESULT
  const coordinates = DEFAULT_COORDINATES
  const vectors = castManifoldVectors<Result, FaceResult>(coordinates)

  describe(".confidenceMetric/2", () => {
    it.each(range(0, vectors.length))("handles vector[%i]", (index: number) => {
      const vector = vectors[index]
      const result = confidenceMetric(face, vector)
      expect(result).toMatchSnapshot()
    })
  })

  describe(".distanceMetric/2", () => {
    it.each(range(0, vectors.length))("handles vector[%i]", (index: number) => {
      const vector = vectors[index]
      const result = distanceMetric(face, vector)
      expect(result).toMatchSnapshot()
    })
  })

  describe(".withinConfidenceBounds/2", () => {
    it("returns true if within bounds (perfect score)", () => {
      const face = {
        ...DEFAULT_USABLE_FACE_RESULT,
        score: 1.0,
        boxScore: 1.0,
        faceScore: 1.0,
      }
      const result = withinConfidenceBounds(face, coordinates)
      expect(result).toEqual(true)
    })

    it("returns true if within bounds", () => {
      const face = {
        ...DEFAULT_USABLE_FACE_RESULT,
        score: 0.9,
        boxScore: 0.9,
        faceScore: 0.9,
      }
      const result = withinConfidenceBounds(face, coordinates)
      expect(result).toEqual(true)
    })

    it("returns false if outside bounds", () => {
      const face = {
        ...DEFAULT_USABLE_FACE_RESULT,
        score: 0.1,
        boxScore: 0.1,
        faceScore: 0.1,
      }
      const result = withinConfidenceBounds(face, coordinates)
      expect(result).toEqual(false)
    })
  })

  describe(".withinToleranceBounds/2", () => {
    it("returns true if within bounds", () => {
      const vector = vectors[0]
      const result = withinToleranceBounds(face, vector)
      expect(result).toEqual(true)
    })

    it("returns false if outside bounds", () => {
      const vector = vectors[1]
      const result = withinToleranceBounds(face, vector)
      expect(result).toEqual(false)
    })

    it.each(range(0, vectors.length))("handles vector[%i]", (index: number) => {
      const vector = vectors[index]
      const result = withinToleranceBounds(face, vector)
      expect(result).toMatchSnapshot()
    })
  })

  describe(".scoreFaceResult/2", () => {
    it.each(range(0, vectors.length))("handles vector[%i]", (index: number) => {
      const vector = vectors[index]
      const result = scoreFaceResult(face, vector)
      expect(result).toMatchSnapshot()
    })
  })
})
