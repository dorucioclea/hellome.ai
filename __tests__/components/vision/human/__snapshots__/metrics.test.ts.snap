// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[0] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[1] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[2] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[3] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[4] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[5] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[6] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[7] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[8] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[9] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[10] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[11] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[12] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[13] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[14] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[15] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[16] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[17] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[18] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[19] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[20] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[21] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[22] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[23] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[24] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[25] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[26] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[27] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[28] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[29] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[30] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[31] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[32] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[33] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .confidenceMetric/2 handles vector[34] 1`] = `0.9702989999999999`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[0] 1`] = `0`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[1] 1`] = `0.15`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[2] 1`] = `0.3`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[3] 1`] = `0.44999999999999996`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[4] 1`] = `0.6`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[5] 1`] = `0.15000000000000002`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[6] 1`] = `0.21213203435596428`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[7] 1`] = `0.33541019662496846`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[8] 1`] = `0.4743416490252569`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[9] 1`] = `0.6184658438426491`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[10] 1`] = `0.30000000000000004`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[11] 1`] = `0.33541019662496846`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[12] 1`] = `0.42426406871192857`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[13] 1`] = `0.5408326913195983`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[14] 1`] = `0.6708203932499369`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[15] 1`] = `0.45`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[16] 1`] = `0.4743416490252569`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[17] 1`] = `0.5408326913195983`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[18] 1`] = `0.6363961030678927`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[19] 1`] = `0.75`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[20] 1`] = `0.6`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[21] 1`] = `0.6184658438426491`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[22] 1`] = `0.6708203932499369`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[23] 1`] = `0.75`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[24] 1`] = `0.848528137423857`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[25] 1`] = `0.75`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[26] 1`] = `0.7648529270389177`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[27] 1`] = `0.8077747210701756`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[28] 1`] = `0.874642784226795`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[29] 1`] = `0.9604686356149273`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[30] 1`] = `0.9`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[31] 1`] = `0.9124143795447329`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[32] 1`] = `0.9486832980505138`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[33] 1`] = `1.0062305898749053`;

exports[`DEFAULT_COORDINATES .distanceMetric/2 handles vector[34] 1`] = `1.0816653826391966`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[0] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[1] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[2] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[3] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[4] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[5] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[6] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[7] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[8] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[9] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[10] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[11] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[12] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[13] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[14] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[15] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[16] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[17] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[18] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[19] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[20] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[21] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[22] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[23] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[24] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[25] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[26] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[27] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[28] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[29] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[30] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[31] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[32] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[33] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .scoreFaceResult/2 handles vector[34] 1`] = `0.99`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[0] 1`] = `true`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[1] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[2] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[3] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[4] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[5] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[6] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[7] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[8] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[9] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[10] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[11] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[12] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[13] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[14] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[15] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[16] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[17] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[18] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[19] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[20] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[21] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[22] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[23] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[24] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[25] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[26] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[27] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[28] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[29] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[30] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[31] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[32] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[33] 1`] = `false`;

exports[`DEFAULT_COORDINATES .withinToleranceBounds/2 handles vector[34] 1`] = `false`;
